src/Execution/Arguments/TypedArgs.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Nuwave\Lighthouse\Execution\Arguments;
+
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Schema\AST\ASTHelper;
+use Nuwave\Lighthouse\Schema\AST\ASTBuilder;
+use GraphQL\Language\AST\FieldDefinitionNode;
+use GraphQL\Language\AST\InputObjectTypeDefinitionNode;
+
+class TypedArgs
+{
+    /**
+     * @var \Nuwave\Lighthouse\Schema\AST\DocumentAST
+     */
+    protected $documentAST;
+
+    /**
+     * @var \Nuwave\Lighthouse\Execution\Arguments\ArgumentTypeNodeConverter
+     */
+    protected $argumentTypeNodeConverter;
+
+    /**
+     * TypedArgs constructor.
+     *
+     * @param  \Nuwave\Lighthouse\Schema\AST\ASTBuilder  $astBuilder
+     * @param  \Nuwave\Lighthouse\Execution\Arguments\ArgumentTypeNodeConverter  $argumentTypeNodeConverter
+     * @return void
+     */
+    public function __construct(ASTBuilder $astBuilder, ArgumentTypeNodeConverter $argumentTypeNodeConverter)
+    {
+        $this->documentAST = $astBuilder->documentAST();
+        $this->argumentTypeNodeConverter = $argumentTypeNodeConverter;
+    }
+
+    /**
+     * Wrap client-given args with type information.
+     *
+     * @param  array  $args
+     * @param  \GraphQL\Type\Definition\ResolveInfo  $resolveInfo
+     * @return \Nuwave\Lighthouse\Execution\Arguments\ArgumentSet
+     */
+    public function fromResolveInfo(array $args, ResolveInfo $resolveInfo): ArgumentSet
+    {
+        $parentName = $resolveInfo->parentType->name;
+        $fieldName = $resolveInfo->fieldName;
+
+        /** @var \GraphQL\Language\AST\ObjectTypeDefinitionNode $parentDefinition */
+        $parentDefinition = $this->documentAST->types[$parentName];
+        /** @var \GraphQL\Language\AST\FieldDefinitionNode $fieldDefinition */
+        $fieldDefinition = ASTHelper::firstByName($parentDefinition->fields, $fieldName);
+
+        return $this->fromField($args, $fieldDefinition);
+    }
+
+    /**
+     * Wrap client-given args with type information.
+     *
+     * @param  array  $args
+     * @param  \GraphQL\Language\AST\FieldDefinitionNode  $fieldDefinition
+     * @return \Nuwave\Lighthouse\Execution\Arguments\ArgumentSet
+     */
+    public function fromField(array $args, FieldDefinitionNode $fieldDefinition): ArgumentSet
+    {
+        $argumentSet = new ArgumentSet();
+        $argumentSet->directives = $fieldDefinition->directives;
+        $argumentSet->arguments = $this->wrapArgs($args, $fieldDefinition->arguments);
+
+        return $argumentSet;
+    }
+
+    /**
+     * Wrap client-given args with type information.
+     *
+     * @param  mixed[]  $args
+     * @param  \GraphQL\Language\AST\NodeList|\GraphQL\Language\AST\InputValueDefinitionNode[] $argumentDefinitions
+     * @return \Nuwave\Lighthouse\Execution\Arguments\Argument[]
+     */
+    protected function wrapArgs(array $args, $argumentDefinitions): array
```

Comment:
I think this method could use some love. It's kinda messy and big.
Just if there are any obvious refactoring you can do here :)

Reply:
Good idea, i broke it down.

Reply:
Yeah, much nicer 👍 Awesome work